### ISSUE TITLE
Logging of password hashes in etcd when transactions timeout

### ISSUE THREAD

**Issue by grepory** [sensu/sensu-go#2342]
## Expected Behavior
We don't log data structures / sanitize data structures before logging so that we're not leaking passwords.

## Current Behavior
During the following test run on CircleCI:

https://circleci.com/gh/sensu/sensu-go/1703

We're logging password hashes when writing to etcd.

```
time="2018-11-15T17:46:24Z" level=warning msg="sync duration of 4.354035354s, expected less than 1s" component=etcd pkg=wal
time="2018-11-15T17:46:31Z" level=error msg="unable to setup agent user" component=backend.seeds error="etcdserver: request timed out"
time="2018-11-15T17:46:31Z" level=warning msg="request \"header:<ID:10788204787227406862 > txn:<compare:<key:\\\"/sensu.io/users/agent\\\" version:0 > success:<request_put:<key:\\\"/sensu.io/users/agent\\\" value:\\\"{\\\\\\\"username\\\\\\\":\\\\\\\"agent\\\\\\\",\\\\\\\"password\\\\\\\":\\\\\\\"[HASH REDACTED]\\\\\\\",\\\\\\\"roles\\\\\\\":[\\\\\\\"agent\\\\\\\"],\\\\\\\"disabled\\\\\\\":false}\\\" > > > \" took too long (7.010072416s) to execute" component=etcd pkg=etcdserver
```

## Possible Solution
Either don't log the structure we're writing or run it through redaction first.

## Steps to Reproduce (for bugs)
Cause an etcd error... Which you could probably do by hammering your disk or provisioning an EBS volume and hammering that to cause IOPS exhaustion.

**Comment by echlebek** [comment]
We should not log the entire structure IMO. It's a bit expensive for a log call.

**Comment by grepory** [comment]
SGTM. I'll make that change.

**Comment by grepory** [comment]
Oh. This is actually a logging message happening _inside_ of etcd. :-/

**Comment by grepory** [comment]
Yup. Right here:

https://github.com/etcd-io/etcd/blob/c9d46ab3799b7f2174268e75f72d01e6d6aac953/etcdserver/util.go#L108-L114

This is still around on master/in later releases. Though it has undergone some changes, it still logs the entire request. I'm not sure what we can do about this. It's at the warning error level in later releases as well, which we want. I wonder if we can get them to change this to a lower level like debug.

**Comment by grepory** [comment]
In master:

https://github.com/etcd-io/etcd/blob/af893d354903175100395a1efb452878f22c6b42/etcdserver/util.go#L140-L164

**Comment by grepory** [comment]
Filed etcd-io/etcd#10264

**Comment by grepory** [comment]
https://github.com/etcd-io/etcd/issues/10264#issuecomment-439155513

**Comment by grepory** [comment]
It looks like this is fixed in etcd 3.3.8. Let's look into updating.

**Comment by palourde** [comment]
We are now running etcd 3.3.13 and from the tests I've made, this is indeed fixed.
